torch/csrc/dynamo/guards.cpp

```diff
@@ -976,6 +976,34 @@ class RelationalGuard : public LeafGuard {
   virtual void reset_state() = 0;
 };
 
+/**
+ * Checks that tensor x is tensor y.
+ */
+class TENSOR_ALIASING : public RelationalGuard {
+ public:
+  TENSOR_ALIASING(py::object verbose_code_parts)
+      : RelationalGuard(verbose_code_parts), _is_first_call(true) {}
+
+  bool check_nopybind(PyObject* value) override { // borrowed ref
+    if (_is_first_call) {
+      _first_tensor = value;
+      _is_first_call = false;
+      return true;
+    }
+    bool result = _first_tensor == value;
+    reset_state();
```

Comment:
We should be able to avoid a virtual method call here for slightly faster performance.